{
  Map<String,CoreContainer.CoreLoadFailure> failures=null;
  Collection<String> cores=null;
  Exception fail=null;
  CoreContainer cc=init(CONFIGSETS_SOLR_XML);
  cores=cc.getCoreNames();
  assertNotNull("core names is null",cores);
  assertEquals("wrong number of cores",0,cores.size());
  failures=cc.getCoreInitFailures();
  assertNotNull("core failures is a null map",failures);
  assertEquals("wrong number of core failures",0,failures.size());
  final CoreDescriptor bogus=new CoreDescriptor(cc,"bogus","bogus_path");
  try {
    ignoreException(Pattern.quote("bogus_path"));
    cc.create(bogus);
    fail("bogus inst dir failed to trigger exception from create");
  }
 catch (  SolrException e) {
    assertTrue("init exception doesn't mention bogus dir: " + e.getCause().getCause().getMessage(),0 < e.getCause().getCause().getMessage().indexOf("bogus_path"));
  }
  cores=cc.getCoreNames();
  assertNotNull("core names is null",cores);
  assertEquals("wrong number of cores",0,cores.size());
  failures=cc.getCoreInitFailures();
  assertNotNull("core failures is a null map",failures);
  assertEquals("wrong number of core failures",1,failures.size());
  fail=failures.get("bogus").exception;
  assertNotNull("null failure for test core",fail);
  assertTrue("init failure doesn't mention problem: " + fail.getCause().getMessage(),0 < fail.getCause().getMessage().indexOf("bogus_path"));
  assertNull(cc.getCore("does_not_exist"));
  try {
    SolrCore c=cc.getCore("bogus");
    fail("Failed to get Exception on accessing core with init failure");
  }
 catch (  SolrException ex) {
    assertEquals(500,ex.code());
    String cause=ex.getCause().getCause().getMessage();
    assertTrue("getCore() ex cause doesn't mention init fail: " + cause,0 < cause.indexOf("bogus_path"));
  }
  cc.shutdown();
}
